expiry/expire.go

```diff
@@ -0,0 +1,39 @@
+package expiry
+
+import (
+	dbEngine "github.com/dicedb/dice/IStorageEngines"
+	"github.com/dicedb/dice/object"
+)
+
+func expireSample(dh dbEngine.IKVStorage) float32 {
+	var limit int = 20
+	var expiredCount int = 0
+	dh.GetStorage().Range(func(k, v interface{}) bool {
+		key := k.(string)
+		value := v.(*object.Obj)
+		// fmt.Printf("The key is : %v and the value is %v\n", key, *value)
+		limit--
+		if object.GetDiceExpiryStore().HasExpired(value) {
+			dh.Del(key)
+			expiredCount++
+		}
+		if limit == 0 {
+			return true
+		}
+		return true
+	})
+	return float32(expiredCount) / float32(limit)
+}
+
+// Deletes all the expired keys - the active way
+// Sampling approach: https://redis.io/commands/expire/
+func DeleteExpiredKeys(dh dbEngine.IKVStorage) {
```

Comment:
let expiry be part of the storage engine implementation and expose it as an interface for an explicit trigger.